origin-bridge/src/utils/attestation.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+
+const eth = require('web3-eth')
+const Web3 = require('web3')
+const web3 = new Web3(process.env.PROVIDER_URL || 'http://localhost:8545')
+const Attestation = require('../models/index').Attestation
+const constants = require('../constants')
+const stringify = require('json-stable-stringify')
+
+async function generateAttestation(
+  attestationType,
+  attestationBody,
+  attestationValue,
+  ethAddress,
+  remoteAddress
+) {
+  const data = {
+    issuer: constants.ISSUER,
+    issueDate: new Date(),
+    attestation: attestationBody
+  }
+
+  const signature = {
+    bytes: generateAttestationSignature(
+      process.env.ATTESTATION_SIGNING_KEY,
+      ethAddress,
+      // Use stringify rather than JSON.stringify to produce deterministic JSON
+      // so the validation of the signature works.
+      stringify(data)
+    ),
+    version: '1.0.0'
+  }
+
+  // Save the attestation in the database
+  await Attestation.create({
+    method: attestationType,
+    ethAddress: ethAddress,
+    value: attestationValue,
+    signature: signature['bytes'],
+    remoteIpAddress: remoteAddress
+  })
+
+  return {
+    schemaId: 'https://schema.originprotocol.com/attestation_1.0.0.json',
+    data: data,
+    signature: signature
+  }
+}
+
+function generateAttestationSignature(privateKey, subject, data) {
+  if (!web3.utils.isHexStrict(privateKey)) {
+    throw 'Invalid private key, not a hex string'
```

Comment:
nit: I think in our code base we tend to throw new Error(<msg>)

Reply:
👍